conda/plugins/virtual_packages/linux.py

```diff
@@ -23,8 +23,11 @@ def conda_virtual_packages():
     # discard everything after the last digit of the third or fourth
     # numeric component; note that this breaks version ordering for
     # development (`-rcN`) kernels, but that can be a TODO for later.
-    _, dist_version = context.platform_system_release
-    dist_version = os.environ.get("CONDA_OVERRIDE_LINUX", dist_version)
+    dist_version = os.environ.get("CONDA_OVERRIDE_LINUX")
+    if dist_version is None:
+        dist_name, dist_version = context.platform_system_release
+        if dist_name != "Linux":
+            dist_version = "0"
     m = re.match(r"\d+\.\d+(\.\d+)?(\.\d+)?", dist_version)
     yield CondaVirtualPackage("linux", m.group() if m else "0", None)
```

Comment:
```suggestion
    if dist_version:
        yield CondaVirtualPackage("linux", m.group() if m else "0", None)
```

This would align behavior with the __osx and __glibc behavior, but there's a test protecting the original implementation. Why? Should we deprecate this and align behaviors?

Reply:
I tried searching through GitHub to find examples of people doing this in the wild but was unsuccessful. That doesn't mean it doesn't happen, but I also think it gives us a valid reason for deprecating current behavior.

I personally cannot understand why if you were going to override the Linux version, you would leave it as `""` if you want it to appear as `0`. In this case, you should just set it to `0`.

Reply:
Yea, same 🤷 Maybe they didn't want to allow `__linux` hides. Or maybe it was an oversight 🤷 A conversation for the [CEP](https://github.com/conda/ceps/pull/103) :D